### PR TITLE
fix #6739 handle unicode in environment variables for py2 activate

### LINKED ISSUE #6739: source activate env error
**I'm submitting a...**
  - [x] bug report
  - [ ] feature request

### Current Behavior
when i try to activate a conda env,this doesn't work for me
the output is:
```
Traceback (most recent call last):
  File "/anaconda/lib/python2.7/site-packages/conda/gateways/logging.py", line 64, in emit
    msg = self.format(record)
  File "/anaconda/lib/python2.7/logging/__init__.py", line 734, in format
    return fmt.format(record)
  File "/anaconda/lib/python2.7/logging/__init__.py", line 465, in format
    record.message = record.getMessage()
  File "/anaconda/lib/python2.7/logging/__init__.py", line 329, in getMessage
    msg = msg % self.args
  File "/anaconda/lib/python2.7/site-packages/conda/__init__.py", line 43, in __repr__
    return '%s: %s' % (self.__class__.__name__, text_type(self))
  File "/anaconda/lib/python2.7/site-packages/conda/__init__.py", line 47, in __str__
    return text_type(self.message % self._kwargs)
ValueError: unsupported format character '{' (0x7b) at index 513
Logged from file exceptions.py, line 724
```

### Steps to Reproduce
i have regularly updated conda to the lateset version recently,
and i installed pipenv.i know that cannot work with a conda-created  env,
but i suspect this may lead to this error
then this error came

##### `conda info`

```
     active environment : None
            shell level : 0
       user config file : /Users/bubu/.condarc
 populated config files : /Users/bubu/.condarc
          conda version : 4.4.7
    conda-build version : 3.2.1
         python version : 2.7.13.final.0
       base environment : /anaconda  (writable)
           channel URLs : https://conda.anaconda.org/anaconda-fusion/osx-64
                          https://conda.anaconda.org/anaconda-fusion/noarch
                          https://repo.continuum.io/pkgs/main/osx-64
                          https://repo.continuum.io/pkgs/main/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
          package cache : /anaconda/pkgs
                          /Users/bubu/.conda/pkgs
       envs directories : /anaconda/envs
                          /Users/bubu/.conda/envs
               platform : osx-64
             user-agent : conda/4.4.7 requests/2.18.4 CPython/2.7.13 Darwin/17.3.0 OSX/10.13.2
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

##### `conda config --show-sources`

```
==> /Users/bubu/.condarc <==
ssl_verify: True
channels:
  - anaconda-fusion
  - defaults
```

##### `conda list --show-channel-urls`

```
# packages in environment at /anaconda:
#
_license                  1.1                      py27_1    defaults
alabaster                 0.7.9                    py27_0    defaults
amqp                      1.4.9                     <pip>
anaconda                  custom                   py27_0    defaults
anaconda-client           1.6.3                    py27_0    defaults
anaconda-navigator        1.5.2                    py27_0    defaults
anaconda-project          0.4.1                    py27_0    defaults
ansible                   2.4.1.0                   <pip>
apipkg                    1.4                       <pip>
appdirs                   1.4.3                     <pip>
appnope                   0.1.0                    py27_0    defaults
appscript                 1.0.1                    py27_0    defaults
argcomplete               1.0.0                    py27_1    defaults
argon2-cffi               16.3.0                    <pip>
astroid                   1.4.9                    py27_0    defaults
astropy                   1.3.3               np112py27_0    defaults
attrs                     17.2.0                    <pip>
autobahn                  17.7.1                    <pip>
Automat                   0.6.0                     <pip>
babel                     2.3.4                    py27_0    defaults
backports                 1.0                      py27_0    defaults
backports.weakref         1.0rc1                    <pip>
backports_abc             0.5                      py27_0    defaults
bcrypt                    3.1.4                     <pip>
beautifulsoup4            4.5.3                    py27_0    defaults
billiard                  3.3.0.23                  <pip>
binaryornot               0.4.3                     <pip>
bitarray                  0.8.1                    py27_0    defaults
blaze                     0.10.1                   py27_0    defaults
bleach                    2.0.0                     <pip>
bokeh                     0.12.4                   py27_0    defaults
boto                      2.45.0                   py27_0    defaults
bottle                    0.12.13                   <pip>
bottle-sqlite             0.1.3                     <pip>
bottleneck                1.2.1               np112py27_0    defaults
ca-certificates           2017.08.26           ha1e5d58_0    defaults
cache-client              0.1.0                     <pip>
cdecimal                  2.3                      py27_2    defaults
celery                    3.1.23                    <pip>
certifi                   2017.11.5        py27hfa9a1c4_0    defaults
certifi                   2017.4.17                 <pip>
cffi                      1.9.1                    py27_0    defaults
Chameleon                 3.1                       <pip>
chardet                   2.3.0                    py27_0    defaults
chardet                   3.0.3                     <pip>
check-manifest            0.35                      <pip>
chest                     0.2.3                    py27_0    defaults
click                     6.7                      py27_0    defaults
click-default-group       1.2                       <pip>
cloudpickle               0.2.2                    py27_0    defaults
clyent                    1.2.2                    py27_0    defaults
colorama                  0.3.7                    py27_0    defaults
conda                     4.4.7                    py27_0    defaults
conda-build               3.2.1                    py27_0    defaults
conda-env                 2.6.0                h36134e3_0    defaults
conda-verify              2.0.0                    py27_0    defaults
configobj                 5.0.6                    py27_0    defaults
configparser              3.5.0                    py27_0    defaults
constantly                15.1.0                    <pip>
contextlib2               0.5.4                    py27_0    defaults
cookiecutter              1.5.1                     <pip>
crockford                 0.0.2                     <pip>
cryptography              1.7.1                    py27_0    defaults
cssselect                 1.0.1                     <pip>
curl                      7.52.1                        0    defaults
cvxopt                    1.1.9                     <pip>
cycler                    0.10.0                   py27_0    defaults
cython                    0.25.2                   py27_0    defaults
cytoolz                   0.8.2                    py27_0    defaults
dask                      0.13.0                   py27_0    defaults
datashape                 0.5.4                    py27_0    defaults
decorator                 4.0.11                   py27_0    defaults
defusedxml                0.5.0                     <pip>
devpi                     2.2.0                     <pip>
devpi-client              3.0.0                     <pip>
devpi-common              3.1.0                     <pip>
devpi-server              4.3.0                     <pip>
devpi-web                 3.2.0                     <pip>
dill                      0.2.5                    py27_0    defaults
Django                    1.10.5                    <pip>
django-extensions         1.7.9                     <pip>
django-filter             1.0.1                     <pip>
django-rest-swagger       2.1.1                     <pip>
django-silk               0.7.3                     <pip>
djangorestframework       3.5.4                     <pip>
djangorestframework-bulk  0.2.1                     <pip>
dlib                      19.4.0                    <pip>
docopt                    0.6.2                     <pip>
docutils                  0.13.1                   py27_0    defaults
dpkt                      1.9.1                     <pip>
dry-rest-permissions      0.1.8                     <pip>
entrypoints               0.2.2                    py27_0    defaults
entrypoints               0.2.2                     <pip>
enum34                    1.1.6                    py27_0    defaults
environment-kernels       1.1                       <pip>
et_xmlfile                1.0.1                    py27_0    defaults
etu-tools                 0.3.0                     <pip>
execnet                   1.4.1                     <pip>
factory-boy               2.8.1                     <pip>
fake-useragent            0.1.7                     <pip>
Faker                     0.7.12                    <pip>
falcon                    1.2.0                     <pip>
fastcache                 1.0.2                    py27_1    defaults
filelock                  2.0.7                    py27_0    defaults
fire                      0.1.2                     <pip>
first                     2.0.1                     <pip>
flake8                    3.3.0                     <pip>
Flask                     0.3                       <pip>
flask                     0.12                     py27_0    defaults
flask-cors                3.0.2                    py27_0    defaults
Flask-Login               0.4.0                     <pip>
freetype                  2.5.5                         2    defaults
funcsigs                  1.0.2                    py27_0    defaults
functools32               3.2.3.2                  py27_0    defaults
future                    0.16.0                    <pip>
futures                   3.0.5                    py27_0    defaults
GB2260                    0.4.1                     <pip>
gdbn                      0.1                       <pip>
get_terminal_size         1.0.0                    py27_0    defaults
gevent                    1.2.2                     <pip>
gevent                    1.2.1                    py27_0    defaults
Glances                   2.9.1                     <pip>
glob2                     0.6              py27h55c9705_0    defaults
gnumpy                    0.2                       <pip>
greenlet                  0.4.12                    <pip>
greenlet                  0.4.11                   py27_0    defaults
grequests                 0.3.0                     <pip>
grin                      1.2.1                    py27_3    defaults
gunicorn                  19.7.1                    <pip>
gym                       0.9.3                     <pip>
h5py                      2.7.0               np112py27_0    defaults
hdf5                      1.8.17                        1    defaults
heapdict                  1.0.0                    py27_1    defaults
hkdf                      0.0.3                     <pip>
holidays                  0.8.1                     <pip>
html5lib                  0.999999999               <pip>
humanize                  0.5.1                     <pip>
hupper                    1.0                       <pip>
hyperlink                 17.3.0                    <pip>
icu                       54.1                          0    defaults
idna                      2.6                       <pip>
idna                      2.2                      py27_0    defaults
imagesize                 0.7.1                    py27_0    defaults
incremental               17.5.0                    <pip>
ipaddress                 1.0.18                   py27_0    defaults
ipykernel                 4.5.2                    py27_0    defaults
ipykernel                 4.6.1                     <pip>
ipython                   5.1.0                    py27_1    defaults
ipython                   5.4.1                     <pip>
ipython_genutils          0.1.0                    py27_0    defaults
ipywidgets                6.0.0                     <pip>
ipywidgets                5.2.2                    py27_1    defaults
isbnlib                   3.7.2                     <pip>
isort                     4.2.5                    py27_0    defaults
itchat                    1.3.10                    <pip>
itsdangerous              0.24                     py27_0    defaults
jbig                      2.1                           0    defaults
jdcal                     1.3                      py27_0    defaults
jedi                      0.10.2                    <pip>
jedi                      0.9.0                    py27_1    defaults
jieba                     0.39                      <pip>
jinja2                    2.9.4                    py27_0    defaults
jinja2-time               0.2.0                     <pip>
joblib                    0.11                      <pip>
jpeg                      9b                            0    defaults
js2xml                    0.3.1                     <pip>
json-lines                0.3.1                     <pip>
jsonschema                2.5.1                    py27_0    defaults
jsonschema                2.6.0                     <pip>
jupyter                   1.0.0                    py27_3    defaults
jupyter-client            5.0.1                     <pip>
jupyter-core              4.3.0                     <pip>
jupyter_client            4.4.0                    py27_0    defaults
jupyter_console           5.0.0                    py27_0    defaults
jupyter_core              4.2.1                    py27_0    defaults
kii-client                0.1.3                     <pip>
kombu                     3.0.37                    <pip>
lazy-object-proxy         1.2.2                    py27_0    defaults
libiconv                  1.14                          0    defaults
libpng                    1.6.27                        0    defaults
libtiff                   4.0.6                         3    defaults
libxml2                   2.9.4                         0    defaults
libxslt                   1.1.29                        0    defaults
line-profiler             2.0                       <pip>
llvmlite                  0.18.0                   py27_0    defaults
locket                    0.2.0                    py27_1    defaults
lxml                      3.7.2                    py27_0    defaults
magic-wormhole            0.10.2                    <pip>
Markdown                  2.2.0                     <pip>
markupsafe                0.23                     py27_2    defaults
matplotlib                2.0.2               np112py27_0    defaults
mccabe                    0.6.1                     <pip>
meld3                     1.0.2                     <pip>
mistune                   0.7.4                     <pip>
mistune                   0.7.3                    py27_1    defaults
mkl                       2017.0.1                      0    defaults
mkl-service               1.1.2                    py27_3    defaults
more-itertools            3.2.0                     <pip>
mpmath                    0.19                     py27_1    defaults
multipledispatch          0.4.9                    py27_0    defaults
mysql                     0.0.1                     <pip>
MySQL-python              1.2.5                     <pip>
nbconvert                 4.2.0                    py27_0    defaults
nbconvert                 5.2.1                     <pip>
nbformat                  4.2.0                    py27_0    defaults
nbformat                  4.3.0                     <pip>
ncmbot                    0.1.6                     <pip>
networkx                  1.11                     py27_0    defaults
nlib                      0.6                       <pip>
nltk                      3.2.2                    py27_0    defaults
nolearn                   0.5b1                     <pip>
nose                      1.3.7                    py27_1    defaults
notebook                  5.0.0                     <pip>
notebook                  4.3.1                    py27_0    defaults
numba                     0.33.0              np112py27_0    defaults
numexpr                   2.6.2               np112py27_0    defaults
numpy                     1.12.1                   py27_0    defaults
numpydoc                  0.6.0                    py27_0    defaults
odo                       0.5.0                    py27_1    defaults
olefile                   0.44                      <pip>
openapi-codec             1.3.2                     <pip>
opencv-python             3.2.0.7                   <pip>
openface                  0.2.1                     <pip>
openpyxl                  2.4.1                    py27_0    defaults
openssl                   1.0.2n               hdbc3d79_0    defaults
org-client                0.4.12                    <pip>
packaging                 16.8                      <pip>
pandas                    0.17.1                    <pip>
pandas                    0.20.2              np112py27_0    defaults
pandocfilters             1.4.1                     <pip>
paramiko                  2.3.1                     <pip>
parse-accept-language     0.1.2                     <pip>
parsel                    1.2.0                     <pip>
partd                     0.3.7                    py27_0    defaults
passlib                   1.7.1                     <pip>
PasteDeploy               1.5.2                     <pip>
path.py                   10.0                     py27_0    defaults
pathlib                   1.0.1                     <pip>
pathlib2                  2.2.0                    py27_0    defaults
patsy                     0.4.1                    py27_0    defaults
pendulum                  1.0.2                     <pip>
pep8                      1.7.0                    py27_0    defaults
petl                      1.1.1                     <pip>
pew                       1.1.2                     <pip>
pexpect                   4.2.1                    py27_0    defaults
pickleshare               0.7.4                    py27_0    defaults
pigar                     0.7.0                     <pip>
pillow                    4.0.0                    py27_0    defaults
pip                       9.0.1            py27h1567d89_4    defaults
pip-tools                 1.9.0                     <pip>
pipreqs                   0.4.9                     <pip>
pkginfo                   1.4.1                     <pip>
pkginfo                   1.4.1                    py27_0    defaults
pluggy                    0.4.0                     <pip>
ply                       3.9                      py27_0    defaults
poyo                      0.4.1                     <pip>
pprofile                  1.11.0                    <pip>
profiling                 0.1.3                     <pip>
progressbar               2.3                       <pip>
prompt_toolkit            1.0.9                    py27_0    defaults
protobuf                  3.3.0                     <pip>
psutil                    5.0.1                    py27_0    defaults
psycopg2                  2.7.1                     <pip>
ptpython                  0.39                      <pip>
ptyprocess                0.5.1                    py27_0    defaults
py                        1.4.32                   py27_0    defaults
py-heat                   0.0.2                     <pip>
py-heat-magic             0.0.2                     <pip>
pyasn1                    0.1.9                    py27_0    defaults
pyasn1-modules            0.0.10                    <pip>
pyaudio                   0.2.7                    py27_0    defaults
pycodestyle               2.3.1                     <pip>
pycosat                   0.6.3            py27h6c51c7e_0    defaults
pycparser                 2.17                     py27_0    defaults
pycrypto                  2.6.1                    py27_4    defaults
pycurl                    7.43.0                   py27_2    defaults
PyDispatcher              2.0.5                     <pip>
pyflakes                  1.5.0                    py27_0    defaults
pyglet                    1.2.4                     <pip>
pygments                  2.1.3                    py27_0    defaults
pylint                    1.6.4                    py27_1    defaults
pymongo                   3.5.1                     <pip>
PyMySQL                   0.7.11                    <pip>
PyNaCl                    1.1.2                     <pip>
pyopenssl                 16.2.0                   py27_0    defaults
pyparsing                 2.1.4                    py27_0    defaults
pyparsing                 2.2.0                     <pip>
pypng                     0.0.18                    <pip>
PyQRCode                  1.2.1                     <pip>
pyqt                      5.6.0                    py27_1    defaults
pyquery                   1.3.0                     <pip>
pyramid                   1.8.3                     <pip>
pyramid-chameleon         0.3                       <pip>
pyresttest                1.7.2.dev0                <pip>
pyspider                  0.3.9                     <pip>
pytables                  3.3.0               np112py27_0    defaults
pytest                    3.0.5                    py27_0    defaults
python                    2.7.13                        0    defaults
python-cjson              1.2.1                     <pip>
python-dateutil           2.6.0                    py27_0    defaults
python-mimeparse          1.6.0                     <pip>
python.app                1.2                      py27_4    defaults
pythonpy                  0.4.11                    <pip>
pytz                      2016.10                  py27_0    defaults
pywavelets                0.5.2               np112py27_0    defaults
pyyaml                    3.12                     py27_0    defaults
pyzmq                     16.0.2                   py27_0    defaults
qiniu                     7.2.0                     <pip>
qt                        5.6.2                         0    defaults
qtawesome                 0.4.3                    py27_0    defaults
qtconsole                 4.2.1                    py27_1    defaults
qtmodern                  0.1.4                     <pip>
qtpy                      1.2.1                    py27_0    defaults
QtPy                      1.3.1                     <pip>
queuelib                  1.4.2                     <pip>
raven                     6.0.0                     <pip>
readline                  6.2                           2    defaults
readme-renderer           17.2                      <pip>
redis                     3.2.0                         0    defaults
redis-py                  2.10.5                   py27_0    defaults
repoze.lru                0.6                       <pip>
requestium                0.1.9                     <pip>
requests                  2.12.4                   py27_0    defaults
requests                  2.18.4                    <pip>
requests-file             1.4.2                     <pip>
rope                      0.9.4                    py27_1    defaults
ruamel_yaml               0.11.14                  py27_1    defaults
scandir                   1.4                      py27_0    defaults
scikit-image              0.13.0              np112py27_0    defaults
scikit-learn              0.18.2              np112py27_0    defaults
scikit-learn              0.17.1                    <pip>
scipy                     0.19.1              np112py27_0    defaults
scipy                     0.16.1                    <pip>
Scrapy                    1.4.0                     <pip>
seaborn                   0.7.1                    py27_0    defaults
selenium                  3.8.0                     <pip>
serpy                     0.1.1                     <pip>
service-identity          17.0.0                    <pip>
setuptools                38.4.0                   py27_0    defaults
setuptools                38.4.0                    <pip>
shutilwhich               1.1.0                     <pip>
simplegeneric             0.8.1                    py27_1    defaults
simplejson                3.11.1                    <pip>
singledispatch            3.4.0.3                  py27_0    defaults
sip                       4.18                     py27_0    defaults
six                       1.10.0                   py27_0    defaults
sklearn                   0.0                       <pip>
slimit                    0.8.1                     <pip>
snowballstemmer           1.2.1                    py27_0    defaults
sockjs-tornado            1.0.3                    py27_0    defaults
spake2                    0.7                       <pip>
speechpy                  1.1                       <pip>
sphinx                    1.5.1                    py27_0    defaults
spyder                    3.1.2                    py27_0    defaults
sqlalchemy                1.1.5                    py27_0    defaults
sqlite                    3.13.0                        0    defaults
ssl_match_hostname        3.4.0.2                  py27_1    defaults
sso-auth-drf              1.4.2                     <pip>
statsmodels               0.8.0               np112py27_0    defaults
subprocess32              3.2.7                    py27_0    defaults
supervisor                3.3.2                     <pip>
sympy                     1.0                      py27_0    defaults
tblib                     1.3.2                     <pip>
tensorflow                1.2.0                     <pip>
terminado                 0.6                      py27_0    defaults
terminaltables            3.1.0                     <pip>
testpath                  0.3.1                     <pip>
textrank4zh               0.3                       <pip>
tflearn                   0.3.2                     <pip>
tk                        8.5.18                        0    defaults
tldextract                2.2.0                     <pip>
toolz                     0.8.2                    py27_0    defaults
tornado                   4.4.2                    py27_0    defaults
tornado                   4.5.1                     <pip>
tox                       2.7.0                     <pip>
tqdm                      4.15.0                    <pip>
traitlets                 4.3.1                    py27_0    defaults
translationstring         1.3                       <pip>
Twisted                   17.5.0                    <pip>
txaio                     2.8.1                     <pip>
txtorcon                  0.19.3                    <pip>
u-msgpack-python          2.4.1                     <pip>
unicodecsv                0.14.1                   py27_0    defaults
urllib3                   1.22                      <pip>
urwid                     1.3.1                     <pip>
valuedispatch             0.0.1                     <pip>
venusian                  1.1.0                     <pip>
virtualenv                15.1.0                    <pip>
virtualenv-clone          0.2.6                     <pip>
w3lib                     1.18.0                    <pip>
waitress                  1.0.2                     <pip>
wcwidth                   0.1.7                    py27_0    defaults
web.py                    0.38                      <pip>
webencodings              0.5.1                     <pip>
WebOb                     1.7.2                     <pip>
weibo                     0.2.2                     <pip>
Werkzeug                  0.12.1                    <pip>
werkzeug                  0.11.15                  py27_0    defaults
wheel                     0.29.0                   py27_0    defaults
whichcraft                0.4.1                     <pip>
Whoosh                    2.7.4                     <pip>
widgetsnbextension        2.0.0                     <pip>
widgetsnbextension        1.2.6                    py27_0    defaults
wrapt                     1.10.8                   py27_0    defaults
WsgiDAV                   2.2.4                     <pip>
xlrd                      1.0.0                    py27_0    defaults
xlsxwriter                0.9.6                    py27_0    defaults
xlwings                   0.10.2                   py27_0    defaults
xlwt                      1.2.0                    py27_0    defaults
xz                        5.2.2                         1    defaults
yaml                      0.1.6                         0    defaults
yarg                      0.1.9                     <pip>
zlib                      1.2.8                         3    defaults
zoo-client                0.7.8                     <pip>
zope.deprecation          4.2.0                     <pip>
zope.interface            4.4.1                     <pip>
```

### PR DESCRIPTION
**Pull request by kalefranz**
fix #6739

### CODE REVIEW

File: conda/activate.py [hunk 1]
```diff
@@ -498,6 +507,15 @@ def ensure_binary(value):
         return value
 
 
+def ensure_unicode(value):
+    try:
+        return value.decode('unicode_escape')
```
Comment (mbargull) [inline conda/activate.py]: I don't think `'unicode_escape'` is correct.
I'd rather assume `"utf-8"` or maybe you'd have to use `sys.getfilesystemencoding()`? (just guessing)

Comment (kalefranz) [reply conda/activate.py]: I just copied this from https://github.com/conda/conda/blob/4.4.7/conda/common/compat.py#L178-L184.  But yeah, maybe actually what we want is https://github.com/conda/conda/blob/4.4.7/conda/common/compat.py#L187-L191

Comment (kalefranz) [reply conda/activate.py]: Or maybe all I really want is [`text_type`](https://github.com/conda/conda/blob/4.4.7/conda/common/compat.py#L37)

Comment (kalefranz) [reply conda/activate.py]: I guess `text_type()` could throw UnicodeDecodeError?  And in that case we should also decode with the filesystem encoding?  I always talk myself in circles with this stuff.

Comment (kalefranz) [reply conda/activate.py]: P.S. @mbargull Can you email me directly? I can't find your contact info.  kfranz at anaconda com

**Comment by kalefranz** [comment]
Think I finally got something that seems sane.  I mean, as sane as python2 on windows dealing with unicode can be.

**Comment by mbargull** [comment]
I believe this looks fine now -- given that I have next to no idea about Python 2's Unicode stuff...
(I'll have to write down my `conda 5.0` wish list some day... Big point on that list: Ditch Python 2 support in `conda` itself 😜 (meaning just don't allow Python 2 in `conda`'s own env but only in separate ones).)

**Comment by kalefranz** [comment]
I'd really like that too, once we have https://github.com/conda/conda/issues/3912.

**Pull request merged**